alacritty_terminal/src/input.rs

```diff
@@ -691,34 +693,34 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
             self.mouse_report(code, ElementState::Released, modifiers);
             return;
         } else if let (Some(point), true) = (point, button == MouseButton::Left) {
-            self.launch_url(modifiers, point);
+            let mouse_state = self.mouse_state(point, modifiers);
+            self.update_mouse_cursor(mouse_state);
+            if let MouseState::Url(url) = mouse_state {
+                let url_bounds = url.linear_bounds(self.ctx.terminal());
+                self.ctx.terminal_mut().set_url_highlight(url_bounds);
+                self.launch_url(url);
+            }
         }
 
         self.copy_selection();
     }
 
     // Spawn URL launcher when clicking on URLs
-    fn launch_url(&self, modifiers: ModifiersState, point: Point) -> Option<()> {
-        if !self.mouse_config.url.mods().relaxed_eq(modifiers)
-            || self.ctx.mouse().block_url_launcher
-        {
-            return None;
+    fn launch_url(&self, url: Url) {
+        if self.ctx.mouse().block_url_launcher {
+            return;
         }
 
-        let point = self.ctx.terminal().visible_to_buffer(point);
-        let url = self.ctx.terminal().urls().drain(..).find(|url| url.contains(point))?;
-        let text = self.ctx.terminal().url_to_string(&url);
-
-        let launcher = self.mouse_config.url.launcher.as_ref()?;
-        let mut args = launcher.args().to_vec();
-        args.push(text);
+        let text = self.ctx.terminal().url_to_string(url);
```

Comment:
This is useless if we don't have url launcher set. So instead we can do it later inside the `if let`.